kind.sh

```diff
@@ -0,0 +1,247 @@
+# kured in it, create a reboot sentinel file, track if reboot is happening
+# by watching the kubernetes api.
+# The resources files/kind cluster are destroyed at the end of the process
+# unless DO_NOT_TEARDOWN is not empty
+# For CI, you should probably set -x before running this script.
+
+#Define the image used in kind, for example:
+#KIND_IMAGE=kindest/node:v1.18.2
+#KIND_IMAGE=kindest/node:v1.18.0@sha256:0e20578828edd939d25eb98496a685c76c98d54084932f76069f886ec315d694
+#KIND_IMAGE=kindest/node:v1.17.0@sha256:9512edae126da271b66b990b6fff768fbb7cd786c7d39e86bdf55906352fdf62
+#KIND_IMAGE=kindest/node:v1.16.4@sha256:b91a2c2317a000f3a783489dfb755064177dbc3a0b2f4147d50f04825d016f55
+KIND_IMAGE=${KIND_IMAGE:-"kindest/node:v1.18.2"}
+
+# desired cluster name; default is "kured"
+KIND_CLUSTER_NAME="${KIND_CLUSTER_NAME:-kured}"
+
+KURED_IMAGE_DEST=${KURED_IMAGE_DEST:-"docker.io/evrardjp/kured:latest"}
+
+#############
+# Functions #
+#############
+
+kubectl="kubectl --context kind-$KIND_CLUSTER_NAME"
+# change nodecount if you change the default kind cluster size
+nodecount=5
+
+cleanup_tmpfiles() {
+	echo "Cleaning up kured tmp folders"
+	rm -rf /tmp/kured-*
+}
+
+teardown_kindcluster() {
+	echo "Tearing down the kind cluster $KIND_CLUSTER_NAME"
+	kind delete cluster --name $KIND_CLUSTER_NAME
+}
+
+cleanup() {
+	if [ -z "$DO_NOT_TEARDOWN" ]; then
+		echo "Tearing down cluster"
+		cleanup_tmpfiles
+		teardown_kindcluster
+        else
+		echo "Gathering cluster logs"
+		kind export logs $tmp_dir/logs --name $KIND_CLUSTER_NAME
+        fi
+}
+
+retry() {
+	set +o errexit
+	local -r -i max_attempts="$1"; shift
+	local -r cmd="$@"
+	local -i attempt_num=1
+
+	until $cmd
+	do
+		if (( attempt_num == max_attempts ))
+		then
+			echo "Attempt $attempt_num failed and there are no more attempts left!"
+			exit 1
+		else
+			echo "Attempt $attempt_num failed! Trying again in $attempt_num seconds..."
+			sleep $(( attempt_num++ ))
+		fi
+	done
+	set -o errexit
+}
+####################
+# Kured build step #
+####################
+
+build_and_push_kuredimage() {
+	make image
+        docker tag docker.io/weaveworks/kured $KURED_IMAGE_DEST
+        docker push $KURED_IMAGE_DEST
+}
+
+############################
+# Kind cluster create step #
+############################
+
+gen_kind_manifest() {
+	echo "Generating kind.yaml"
+	# Please don't name your kind nodes ".*true.*" or "<none>", if you name them
+	cat <<EOF > $tmp_dir/kind.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  image: $KIND_IMAGE
+- role: control-plane
+  image: $KIND_IMAGE
+- role: control-plane
+  image: $KIND_IMAGE
+- role: worker
+  image: $KIND_IMAGE
+- role: worker
+  image: $KIND_IMAGE
+EOF
+}
+
+check_all_nodes_are_ready() {
+	$kubectl get nodes | grep Ready | wc -l | grep $nodecount > /dev/null
+}
+
+spinup_cluster() {
+	gen_kind_manifest
+	kind create cluster --name "${KIND_CLUSTER_NAME}" --config=$tmp_dir/kind.yaml
```

Comment:
Maybe first check if `kind` is installed?

Reply:
ok.